### PR TITLE
Add the ability to ignore changes to desired size of node groups

### LINKED ISSUE #985: Ability to Ignore Changes on `desiredSize` in eks.ManagedNodeGroup
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

### Context
Currently, any change to the `desiredSize` of a node group leads to updates within the EKS ManagedNodeGroup and the child NodeGroup, which can be disruptive and undesired in certain scenarios. For example, when scaling operations are managed by external tools or autoscaling policies, the Pulumi should not interfere with these dynamic changes.

### Affected area/feature

eks.ManagedNodeGroup in AWS EKS

### Expected Behavior
Introduce a configuration option, possibly a boolean flag, that allows the `desiredSize` of a node group to be ignored during updates. This would ensure that changes to this specific property do not trigger updates in the EKS ManagedNodeGroup, providing greater flexibility in managing node group sizes.

### Impact
This feature would greatly enhance the usability of eks.ManagedNodeGroup for scenarios involving dynamic scaling, making it more adaptable to various operational requirements.

### PR DESCRIPTION
**Pull request by flostadler**
When using [Cluster Autoscaler](https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler) for automatically scaling node groups based on cluster requirements pulumi permanently shows a diff.

The problem is because the cluster autoscaler takes control of the desiredSize of the scalingConfig, putting it out of sync with Pulumi state.

MLCs don't support ignore changes for their children resources, because of this we're adding a new input that allows you to selectively ignore scaling changes.
This is not added to the deprecated `NodeGroup` (aka `NodeGroupV1`) because the desired size is part of the cloudformation stack that is deployed as part of that component. We cannot selectively ignore changes to the cloudformation template because its a string.

Closes #985 #1293

### CODE REVIEW

File: provider/test-programs/ignore-scaling-changes/index.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+export const kubeconfig = cluster.kubeconfig;
+
+// Create a simple AL 2023 node group with x64 instances
+const mng = eks.createManagedNodeGroup("ignore-scaling-config-mng", {
+  scalingConfig: {
+    minSize: 1,
+    maxSize: 3,
+    desiredSize,
+  },
+  ignoreScalingChanges: true,
```
Comment (t0yv0) [inline provider/test-programs/ignore-scaling-changes/index.ts:48]: From which parameters here does it follow that this resource is using "cluster autoscaler"? If so could ignoreScalingChanges: true be inferred automatically?

File: provider/test-programs/ignore-scaling-changes/index.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+const ngV2 = new eks.NodeGroupV2("ignore-scaling-config-ngv2", {
+  minSize: 1,
+  maxSize: 3,
+  desiredCapacity: desiredSize,
+  cluster: cluster,
+  instanceType: "t3.medium",
+  operatingSystem: eks.OperatingSystem.AL2023,
+  instanceProfile: new aws.iam.InstanceProfile("ignore-scaling-config-ngv2", {role: ngV2Role}),
+  ignoreScalingChanges: true,
```
Comment (t0yv0) [inline provider/test-programs/ignore-scaling-changes/index.ts:63]: Same q here.

**Review by t0yv0 — APPROVED** [review]
Could also support this by directing users to the transforms resource option, but it's a little more involved so this seems nice.   Left some questions on opt-in vs auto-infer.

File: provider/test-programs/ignore-scaling-changes/index.ts (hunk 1 above)
Comment (flostadler) [reply provider/test-programs/ignore-scaling-changes/index.ts:48]: Cluster Autoscaler is not an EKS feature, it's software users can deploy in their clusters.

**Comment by flostadler** [comment]
> Could also support this by directing users to the transforms resource option, but it's a little more involved so this seems nice.   Left some questions on opt-in vs auto-infer. 

I think we shouldn't require users to meddle with transforms for common use-cases. Cluster autoscaling is one of them

**Pull request merged**